src/python/pants/backend/python/subsystems/pytest.py

```diff
@@ -44,6 +44,27 @@ def register_options(cls, register):
              help='Requirements string for the unittest2 library, which some python versions '
                   'may need.',
              removal_version="1.25.0.dev0", removal_hint="Use --pytest-plugins instead.")
+    register(
+      '--timeout-disabled',
```

Comment:
@jsirois @stuhood @benjyw @blorente  any opposition that this uses `--timeout-disabled` rather than `--timeouts`? 

The main motivation was to use the same naming scheme as `--timeout-default` and `--timeout-maximum`.

Then, our thinking was that the main use case for this option is when people want to turn off timeouts when iterating on code, so it's more natural to write `--timeout-disabled` than `--no-timeout-enabled`.

Reply:
I personally like `--no-timeout-enabled` better, but it might just be because I'm so conditioned by the way pants options work that I'd expect that `--no` to be there, it feels more consistent.
These were my two bikeshed cents, feel free to ignore them.

Reply:
I agree with Borja, but I'd go with `--no-timeout` or `--no-timeouts` which I think you don't like. To me, it says disable all the `--timeout-*` options and that is in fact what it does. Plus, more concise.

Reply:
`—no-timeout` sounds good to me. Yes, I don’t like the s with `—no-timeouts`.

Reply:
`—no-timeout` sounds good to me. Yes, I don’t like the s with `—no-timeouts`.

Reply:
`--no-timeout` is not possible until the deprecation cycle finishes with https://github.com/pantsbuild/pants/pull/8805, which will be well past landing this PR. So..are we thinking `--no-timeouts` or `--no-timeout-enabled`?

I prefer the latter for the consistent name scheme with the other `--timeout` options but acknowledge the added verbosity.

Reply:
Would it be worth marking the option here with the same deprecation cycle (or perhaps one step after) as in #8805 now to be able to quickly transition to consistency with other `--timeout` options?

Reply:
We can't start a deprecation until the new, correct behavior is possible. So, this means that we would need to use `--no-timeouts` (or an alternative) until 1.26.0.dev2, and then we could introduce `--no-timeout` and deprecate `--no-timeouts` to be removed in 1.28.0.dev2. I'm fine with that, personally.

Reply:
I think I'm leaning towards `--no-timeouts` for now, to keep it in line with V1. Once the deprecation cycle from #8805 is done we can update both the v1 task and v2 rule to something that makes more sense.